drivers/sx1276/sx1276.c

```diff
@@ -0,0 +1,1302 @@
+    }
+
+    return true;
+}
+
+bool sx1276_is_channel_free(sx1276_t *dev, uint32_t freq, int16_t rssi_thresh)
+{
+    int16_t rssi = 0;
+
+    sx1276_set_channel(dev, freq);
+    sx1276_set_op_mode(dev, SX1276_RF_OPMODE_RECEIVER);
+
+    xtimer_usleep(1000); /* wait 1 millisecond */
+
+    rssi = sx1276_read_rssi(dev);
+    sx1276_set_sleep(dev);
+
+    if (rssi > rssi_thresh) {
+        return false;
+    }
+
+    return true;
+}
+
+void sx1276_set_modem(sx1276_t *dev, sx1276_radio_modems_t modem)
+{
+    dev->settings.modem = modem;
+
+    switch (dev->settings.modem) {
+        case SX1276_MODEM_LORA:
+            sx1276_set_op_mode(dev, SX1276_RF_OPMODE_SLEEP);
+            sx1276_reg_write(dev,
+                             SX1276_REG_OPMODE,
+                             (sx1276_reg_read(dev, SX1276_REG_OPMODE)
+                              & SX1276_RF_LORA_OPMODE_LONGRANGEMODE_MASK)
+                             | SX1276_RF_LORA_OPMODE_LONGRANGEMODE_ON);
+
+            sx1276_reg_write(dev, SX1276_REG_DIOMAPPING1, 0x00);
+            sx1276_reg_write(dev, SX1276_REG_DIOMAPPING2, 0x10); /* DIO5=ClkOut */
+            break;
+
+        case SX1276_MODEM_FSK:
+            sx1276_set_op_mode(dev, SX1276_RF_OPMODE_SLEEP);
+            sx1276_reg_write(dev,
+                             SX1276_REG_OPMODE,
+                             (sx1276_reg_read(dev, SX1276_REG_OPMODE)
+                              & SX1276_RF_LORA_OPMODE_LONGRANGEMODE_MASK)
+                             | SX1276_RF_LORA_OPMODE_LONGRANGEMODE_OFF);
+
+            sx1276_reg_write(dev, SX1276_REG_DIOMAPPING1, 0x00);
+            //sx1276_reg_write(dev, SX1276_REG_DIOMAPPING2, 0x20); /* DIO5=mode_ready */
+            break;
+        default:
+            break;
+    }
+}
+
+#define RXLORA_RXMODE_RSSI_SX1276_REG_MODEM_CONFIG1 0x0A
+#define RXLORA_RXMODE_RSSI_SX1276_REG_MODEM_CONFIG2 0x70
+
+uint32_t sx1276_random(sx1276_t *dev)
+{
+    uint8_t i;
+    uint32_t rnd = 0;
+
+    sx1276_set_modem(dev, SX1276_MODEM_LORA); /* Set LoRa modem ON */
+
+    /* Disable LoRa modem interrupts */
+    sx1276_reg_write(dev, SX1276_REG_LR_IRQFLAGSMASK, SX1276_RF_LORA_IRQFLAGS_RXTIMEOUT |
+                     SX1276_RF_LORA_IRQFLAGS_RXDONE |
+                     SX1276_RF_LORA_IRQFLAGS_PAYLOADCRCERROR |
+                     SX1276_RF_LORA_IRQFLAGS_VALIDHEADER |
+                     SX1276_RF_LORA_IRQFLAGS_TXDONE |
+                     SX1276_RF_LORA_IRQFLAGS_CADDONE |
+                     SX1276_RF_LORA_IRQFLAGS_FHSSCHANGEDCHANNEL |
+                     SX1276_RF_LORA_IRQFLAGS_CADDETECTED);
+
+    sx1276_set_op_mode(dev, SX1276_RF_OPMODE_STANDBY);
+    sx1276_reg_write(dev, SX1276_REG_LR_MODEMCONFIG1, RXLORA_RXMODE_RSSI_SX1276_REG_MODEM_CONFIG1);
+    sx1276_reg_write(dev, SX1276_REG_LR_MODEMCONFIG1, RXLORA_RXMODE_RSSI_SX1276_REG_MODEM_CONFIG2);
+
+    /* Set radio in continuous reception */
+    sx1276_set_op_mode(dev, SX1276_RF_OPMODE_RECEIVER);
+
+    for (i = 0; i < 32; i++) {
```

Comment:
32 is the number of random bits that we will get from the transceiver and then we will put them into 32 bit integer.